pkgs/firehose/CHANGELOG.md

```diff
@@ -1,3 +1,6 @@
+## 0.3.20
```

Comment:
Done, and got an error: "error: pubspec version (0.3.20-wip) and changelog (0.3.20) don't agree"

I've been thinking about that. I originally thought it was a bug.
It's probably correct, and the CHANGELOG.md should be updated to `## 0.3.20-wip`.

Reasoning: When working on a package, between releases, every change may force an increment over the previous release. It's important to track that, but also to track, unambiguously, what the most recent release version was, so that I won't do two increments of the minor version just because I do two individual changes between releases. Only increment minor version if it's not already incremented, which you only know if you know what it was before being incremented the first time.

By having `-wip` on the CHANGELOG entry until release, we can use the most recent non `-wip` entry in the changelog to find the version of the most recent release.

It does mean you have to remove the `-wip` from both pubspec.yaml and CHANGELOG.md before releasing (but this precise validation will catch it if they differ, so it's doing good work here.)


So I guess I'd want a tool to do the updating for me:
* `repo-version -wip +patch` ensures the current version is at least a patch increment over the most recent release, and adds `--wip` to the version in both CHANGELOG.md and pubspec.yaml.
* `repo-version +minor` ensures the current version is at least a minor increment over the most recent release, and removes any pre-release tags. 
* `repo-version -wip` adds `-wip` to the current version again. In case you forgot it in the line above.

Both keep the CHANGELOG and pubspec in sync, and *add* an entry to the changelog if there isn't one for the current pubspec version.

Is that a tool we already have?

Reply:
Yeah, so the tool currently just validates that the changelog and pubspec versions exactly agree. `-wip` indicates that the changes in that section haven't been published. As a manual process (by review or other), we do want a new entry for every user-facing change. A new section has to be added if it's the first change after a publish. We'll drop the `-wip` suffix once we're ready to publish what we have (we may batch up several changes, or just publish on the first change depending on various factors).

There probably _is_ more tooling we could do here; things like helping you create the content of the PR (adding a new changelog entry; rev'ing the pubspec version). That would be independent of general PR validation. Is that what you were thinking of?